platforms/hyperledger-fabric/configuration/roles/create/channel_artifacts/tasks/main.yaml

```diff
@@ -76,6 +76,34 @@
   tags:
     - molecule-idempotence-notest
 
+- name: "Check if genesis block has already been written to Vault"
+  shell: |
+    vault kv get -field=genesisBlock_{{ channel_name }} secret/crypto/ordererOrganizations
+  environment:
+    VAULT_ADDR: "{{ org.vault.url }}"
+    VAULT_TOKEN: "{{ org.vault.root_token }}"
+  register: vault_genesis_result
+  ignore_errors: yes
+  changed_when: false
+  loop: "{{ network['organizations'] }}"
+  loop_control:
+    loop_var: org
+  when: org.type == "orderer"
+
+- name: "Write genesis block to Vault"  
+  shell: |
+    vault write secret/crypto/ordererOrganizations genesisBlock_{{ channel_name }}="$(cat {{build_path}}/channel-artifacts/genesis.block.base64)"
+  environment:
+    VAULT_ADDR: "{{ org.vault.url }}"
+    VAULT_TOKEN: "{{ org.vault.root_token }}"
+  when: add_new_org == 'false' and vault_genesis_result.failed == False 
```

Comment:
why this condition? if vault_genesis_result.failed is false, the secret is already present in Vault

Reply:
I got this condition from another file. Wasn't able to find any solution for this online. See snippet below
https://github.com/hyperledger-labs/blockchain-automation-framework/blob/81290c6421adb3cf5179834eb1a5fbd02cfeb7d4/platforms/hyperledger-fabric/configuration/roles/create/crypto/peer/tasks/main.yaml#L132-L155

Reply:
vault_genesis_result.failed == True

Reply:
or just vault_genesis_result.failed

Reply:
How does this interact with the `ignore_errors` flag? Will the result still be saved?

Reply:
also there is 2 when in the task 105 and 99

Reply:
Resolved in [ad25b2c](https://github.com/hyperledger-labs/blockchain-automation-framework/pull/1053/commits/ad25b2c131b4539efa614b1f5f263c972b7bf629)